### PR TITLE
Unmanned camera drone vehicle

### PR DESCRIPTION
**Pull request by tornikeo**
#### Summary
Content "Add controllable camera drones to game"

#### Purpose of change

To make bird mutants seethe.

And also to be able to scan your environment with these things. Related issue #71564

### CODE REVIEW

**Comment by tornikeo** [comment]
For some odd reason, the vehicle doesn't unfold correctly, `bad vehicle part` error. Still trying to catch what's wrong.

**Comment by fairyarmadillo** [comment]
This isn't going to work. If you read the issue this came from, the view is locked to the player and you can't really change that without changing how the reality bubble works.

File: data/json/items/vehicle/frames.json [hunk 1]
```diff
@@ -32,6 +32,15 @@
     "description": "A small, foldable lightweight frame made from aluminum.",
     "copy-from": "xlframe"
   },
+  {
+    "type": "GENERIC",
+    "id": "fold_flying_drone_frame",
+    "name": { "str": "foldable flying drone frame" },
+    "description": "A small, foldable lightweight frame made from aluminum.",
+    "material": [ "plastic", "carbonfiber" ],
```
Comment (db48x) [inline data/json/items/vehicle/frames.json:40]: Description is wrong, or the materials list is.

**Comment by Inglonias** [comment]
I love the idea, but I agree with fairy on this. We just don't have the technology right now to make this reasonable. You simply have no way to track the drone when it's on other Z-levels.

**Comment by esotericist** [comment]
> I love the idea, but I agree with fairy on this. We just don't have the technology right now to make this reasonable. You simply have no way to track the drone when it's on other Z-levels.

this is trivially solvable. i promise you, it won't be hard.

**Comment by ashGlaw** [comment]
I could see abstracting it, making it an item that takes X period of time to use and once you do, it reveals an area of the map around you, at the cost of the energy that it takes to pilot the drone

**Comment by kevingranade** [comment]
> view is locked to the player and you can't really change that without changing how the reality bubble works

You absolutely can,  it's not even that hard. It might even be a simple as hacking the vehicle camera code to allow it do work when controlling a vehicle remotely.

**Comment by Inglonias** [comment]
> > view is locked to the player and you can't really change that without changing how the reality bubble works
> 
> You absolutely can,  it's not even that hard. It might even be a simple as hacking the vehicle camera code to allow it do work when controlling a vehicle remotely. 

This leads to the possibility of the player being killed off screen while they control their drone. That situation would have to be tested for.

**Comment by esotericist** [comment]
> > > view is locked to the player and you can't really change that without changing how the reality bubble works
> > 
> > 
> > You absolutely can,  it's not even that hard. It might even be a simple as hacking the vehicle camera code to allow it do work when controlling a vehicle remotely.
> 
> This leads to the possibility of the player being killed off screen while they control their drone. That situation would have to be tested for.

this is no different than the possibility of being killed while aiming at a distant target, or reading, or crafting. we already have tools for this, and established expectations for "how safe am i if i do something attention-consuming in the open"

find a secluded spot to launch your drone from, make sure safe mode is configured right, and you don't have distractions ignored.

**Comment by db48x** [comment]
Yep. I nearly got killed by a wasp while working on my canoe the other day, and later one of my NPC friends got killed by a zombie right outside my kitchen while I was inside cooking. No prompt to stop cooking when combat started either, I just found a dozen rounds of combat noises in the log once I finished roasting those all–important acorns. I don’t have a problem at all with the danger of being killed while operating a drone. That is entirely in keeping with the theme of the game.

That said, it would be especially cool to have a split–screen view while controlling a vehicle remotely. Or even just show a 5×5 view of the area right around the player down in the corner of the screen, instead of splitting the screen 50/50. Work for the future.

**Comment by Consoleable** [comment]
I'm excited! I bet you can overcome any technical issues if you keep working on this.

**Pull request closed without merging**